src/dllist.c

```diff
@@ -995,6 +995,97 @@ static PyObject* dllist_popright(DLListObject* self)
     return value;
 }
 
+static PyObject* dllist_pop(DLListObject* self, PyObject *arg)
+{
+    DLListNodeObject *del_node;
+    DLListNodeObject *prev_node;
+
+    PyObject *value;
+    PyObject *indexObject = NULL;
+
+    Py_ssize_t index;
+    Py_ssize_t i;
+
+
+    if (!PyArg_UnpackTuple(arg, "pop", 0, 1, &indexObject))
+    {
+        return NULL;
+    }
+
+    /* If we did not get passed the "index" arg, just return popright */
+    if ( indexObject == NULL )
+    {
+        return dllist_popright( self );
+    }
+
+    if (self->first == Py_None)
+    {
+        PyErr_SetString(PyExc_ValueError, "List is empty");
+        return NULL;
+    }
+    else
+    {
+        if (!Py23Int_Check(indexObject))
+        {
+            PyErr_SetString(PyExc_TypeError, "Index must be an integer");
+            return NULL;
+        }
+
+        index = Py23Int_AsSsize_t(indexObject);
+    }
+
+
+    /* Negative index */
+    if (index < 0)
+        index = ((DLListObject*)self)->size + index;
+
+    /* Either a negative greater than index size, or a positive greater than size */
+    if ( index < 0 || index >= ((DLListObject*)self)->size )
```

Comment:
You might reuse `dllist_get_node_internal` here to locate the node to remove. This function can walk the list from the last accessed element, so the following code could be faster:
```
item = list[idx]    # remembers last acessed item
list.pop(idx)
```